evaluations/subcoin_2_piewol.md

```diff
@@ -0,0 +1,191 @@
+# Evaluation
+
+- **Status:** in progress
+- **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/Subcoin.md
+- **Milestone:** 2
+
+| Number | Deliverable    | Accepted | Link | Notes |
+| -- | --  | ---    | --- | --- |
+| 0a.    | License                 | <ul><li>[x] </li></ul> | https://github.com/subcoin-project/subcoin/blob/subcoin-milestone-2/LICENSE |  --  |
+| 0b.    | Documentation           | <ul><li>[ ] </li></ul> | Each module has its own docs. The rendered inline rustdoc is deployed at https://subcoin-project.github.io/subcoin/.  | some elements e.g. structs are still without inline doc |
+| 0c.    | Testing and Testing Guide | <ul><li>[x] </li></ul> |  https://github.com/subcoin-project/subcoin/tree/subcoin-milestone-2?tab=readme-ov-file#run-tests |  see general notes  |
+| 0d.    | Docker                  | <ul><li>[ ] </li></ul> | https://github.com/subcoin-project/subcoin/blob/subcoin-milestone-2/Dockerfile The docker image is available at https://github.com/subcoin-project/subcoin/pkgs/container/subcoin/249545041?tag=v0.2.0  |  builds. see notes. |
+| 1.     | Block Verification	       | <ul><li>[ ] </li></ul> | 	https://github.com/subcoin-project/subcoin/blob/subcoin-milestone-2/crates/sc-consensus-nakamoto/src/verification.rs |  how comes that there are no tests? Is this directly from ``rust-bitcoin``? |
+| 2.     | `bitcoin-network`       | <ul><li>[x] </li></ul> | https://github.com/subcoin-project/subcoin/blob/subcoin-milestone-2/crates/subcoin-network/src/lib.rs | works, gets peers and imports blocks  |
+
+# General Notes
+
+
+In ``pallet-bitcoin`` you use the error message "txid must be encoded correctly; qed" both for encoding and decoding operations. I think it would be great it these error messages were unique for both operations. Furthermore it looks like the result type in the trait is used for a reason as the writer can emit errors. Whats the reason you put an `.expect()` here allowing for a panic? Basically, why are you sure that it will never happen? 
```

Comment:
Since `pallet-bitcoin` contains only one unsigned extrinsic, which is _constructed internally, plus the Rust type `bitcoin::Txid` ensures the Bitcoin encode/decode works (guaranteed by `rust-bitcoin`), therefore the panic never happens.